### ISSUE TITLE
SetColStyle does not work in cells where a time value is present before the style is applied

### ISSUE THREAD

**Issue by remoremorali** [360EntSecGroup-Skylar/excelize#695]
**Description**
When setting a style with SetColStyle, the cells that already contain a date value are not formatted.
The strings and numbers work correctly, and time values added after setting the style are formatted correctly.
SetCellStyle always works.

**Steps to reproduce the issue:**
```
func TestSetColRepro(*testing.T) {
	xls := NewFile()
	t := time.Now()
	t = t.UTC()
	header := []interface{}{"Str Col", "Num Col", "Num Cell", "Num NoStyle", "Date Col", "Date Col", "Date Cell", "Date NoStyle"}
	xls.SetSheetRow("Sheet1", "A1", &header)
	slice := []interface{}{"Test", -12.34, -12.34, -12.34, t, t, t, t, "This row was set before applying the styles"}
	xls.SetSheetRow("Sheet1", "A2", &slice)

	style, _ := xls.NewStyle(`{"fill":{"type":"pattern","color":["#E0EBF5"],"pattern":1}, "number_format": 14}`)
	xls.SetColStyle("Sheet1", "A:B", style)
	xls.SetColStyle("Sheet1", "E:F", style)
	xls.SetCellStyle("Sheet1", "C2", "C3", style)
	xls.SetCellStyle("Sheet1", "G2", "G3", style)

	slice2 := []interface{}{"Test", -12.34, -12.34, -12.34, t, t, t, t, "This row was set after applying the styles"}
	xls.SetSheetRow("Sheet1", "A3", &slice2)
	xls.SaveAs("/tmp/repro.xls")
}
```
**Describe the results you received:**
![Results](https://i.imgur.com/eshZYaF.png)
Col/Cell/No style refer to the kind of SetXXXStyle applied

Cells E2 and F2 are not formatted according to the applied style
Cells E3 and E4, added after applying the style are formatted correctly

**Describe the results you expected:**
All the cells should be formatted correctly

**Output of `go version`:**

`go version go1.13.8 linux/amd64`

**Excelize version or commit ID:**

`3c8c8c5` (master/HEAD)

**Environment details (OS, Microsoft Excel™ version, physical, etc.):**
Ubuntu 20.04
Libreoffice Version: 6.4.4.2
Google sheets has the same issue

**Comment by remoremorali** [comment]
I dug a little bit into this issue: `SetCellValue` calls `SetCellTimeFunc` which calls [`SetDefaultTimeStyle`](https://github.com/360EntSecGroup-Skylar/excelize/blob/3c8c8c55c8128c5bb94fe28451f58fbc5fb4a118/cell.go#L150) to only set a style that displays the number as a date if there is no other style yet.
There is no style set on the cell, so a default time style gets applied.

When adding a date value after setting the column style, `GetCellStyle` will return the column style which won't be overridden.

**Comment by remoremorali** [comment]
Ultimately the function that drives the style is [`prepareCellStyle`](https://github.com/360EntSecGroup-Skylar/excelize/blob/3c8c8c55c8128c5bb94fe28451f58fbc5fb4a118/cell.go#L774) which will use the column style only if the cell has no style set.

This means that any cell that had a custom style set won't have it changed when the column style is applied, which I think it's not what happens in the regular spreadsheet apps.

I see two solutions for this:

1. When `SetColStyle` is called, it can go over the existing cells in the column and set the style explicitly
2. Track the "default time style" and in `prepareCellStyle` treat it in the same way as if style is not set.

Solution 1 surely works but can be slow when there are a lot of rows; otoh it also sets the new style when a custom style has been set. It could also be a breaking change for existing behavior.
Solution 2 is more complex to implement. 
  * One way would be to always create a style at the beginning so it has a standard value and use that one, but I am not sure what happens when loading a file that already has styles: one would have to handle the conflicts. And this solution would not work to override any pre-existing cell style.

@xuri wdyt?

**Comment by remoremorali** [comment]
Gentle ping: I'd be happy to propose a fix, but I'd like to know if my proposed solution is ok. @xuri wdyt?
Thanks

**Comment by xuri** [comment]
Sorry for my late reply, this issue should be fixed after commit [4f0d676](https://github.com/360EntSecGroup-Skylar/excelize/commit/4f0d676eb765472d1fe7a29cacd165b982785bd2), please try to upgrade the master branch code, and this patch will be released in the next version.
